### PR TITLE
Test PR

### PR DESCRIPTION
**Pull request by anto2156**

### CODE REVIEW

**Comment by michaelkirk** [comment]
Since this is marked as a draft, I'm not looking too closely yet. Let me or @dabreegster know if you have any questions.

**Comment by dabreegster** [comment]
We discussed the approach in Slack; copy/paste isn't formatting super well, but:

> So IIUC, you're plumbing TripParameters  as a modifier on a Scenario. I think it might be easier to have the parameters one level "higher" -- use TripParameters to generate the Scenario entirely

> I hadn't anticipated having an edit button once you start the census-based scenario, but I think it makes lots of sense. The user flow would be:

> 1)    Open the traffic dropdown
> 2)    Choose census
> 3)    A scenario using Config::default() is generated, then fed to... I guess PlayScenario
> 4)    There's a button to edit the parameters of the census thing, clicking it brings up the new modal
> 5)   Saving from that modal calls popdat::generate_scenario again with the new settings

> The question is how to remember Config or TripParameters or whatever we call it. Just keeping the scenario around would lose that

> I guess it's a little similar to how PlayScenaro remembers the modifiers. It'd be kind of weird to store the modifiers on a scenario itself, so that's why PlayScenario keeps around the two things it needs to do it again from scratch


Not sure if that was coherent, but in short, I think there's a simpler approach: forking `PlayScenario` and storing `popdat::Config` in the state, so the player can generate a scenario, tune the parameters, regenerate, repeat

**Comment by michaelkirk** [comment]
ah, great thanks for the context @dabreegster!

File: game/src/sandbox/gameplay/census.rs [hunk 1]
```diff
@@ -38,15 +55,19 @@ impl State<App> for CensusGenerator {
                 }
                 "Generate" => {
                     // Generate this from self.panel
-                    let config = Config::default();
+                    let mut config = Config::default();
+                    let walk_shorter_than = self.panel.spinner("walk shorter than") as usize;
+                    let walk_bike_shorter_than = self.panel.spinner("walk bike shorter than") as usize;
+
+                    config.walk_for_distances_shorter_than = Distance::miles((walk_shorter_than as f64 / 10 as f64) as f64);
```
Comment (dabreegster) [inline game/src/sandbox/gameplay/census.rs:62]: Lots of casting! `panel.spinner()` hands back an `isize` (signed integer), so I think you could immediately cast to `f64` a few lines up, then use `10.0` to tell Rust the constant is floating pt, and then you don't need the final cast. So this line should just wind up being `Distance::miles(walk_shorter_than / 10.0)`

**Comment by dabreegster** [comment]
I'm going to close this to clean up. We're not actively working on generating census-based demand models right now, and we have to revisit what parameters should even be controllable from the UI. Thanks again for working on this!

**Pull request closed without merging**